src/core/server/models/story/story.ts

```diff
@@ -727,3 +735,124 @@ export async function retrieveStorySections(
   // the null entries, the resulting array can not contain null.
   return results.filter((section) => section !== null).sort() as string[];
 }
+
+export async function markStoryForArchiving(
+  mongo: Db,
+  tenantID: string,
+  id: string,
+  now: Date
+) {
+  const result = await collection(mongo).findOneAndUpdate(
+    {
+      id,
+      tenantID,
+      isArchiving: { $in: [null, false] },
+      isArchived: { $in: [null, false] },
+    },
+    {
+      $set: {
+        isArchiving: true,
+        closedAt: now,
+        updatedAt: now,
+        startedArchivingAt: now,
+      },
+    },
+    {
+      returnOriginal: false,
+    }
+  );
+
+  return result.value;
+}
+
+export async function markStoryForUnarchiving(
+  mongo: Db,
+  tenantID: string,
+  id: string,
+  now: Date
+) {
+  const result = await collection(mongo).findOneAndUpdate(
+    {
+      id,
+      tenantID,
+      isArchiving: false,
+      isArchived: true,
+    },
+    {
+      $set: {
+        isArchiving: true,
+        closedAt: now,
+        updatedAt: now,
+        startedUnarchivingAt: now,
+      },
+    },
+    {
+      returnOriginal: false,
+    }
+  );
+
+  return result.value;
+}
+
+export async function retrieveStoriesToBeArchived(
+  mongo: Db,
+  tenantID: string,
+  olderThan: Date
+): Promise<Cursor<Readonly<Story>>> {
+  const stories = collection(mongo).find({
+    tenantID,
+    createdAt: { $lte: olderThan },
+    isArchiving: { $in: [null, false] },
+    isArchived: { $in: [null, false] },
+    startedUnarchivingAt: { $in: [null, false] },
+    unarchivedAt: { $in: [null, false] },
+  });
+
+  return stories;
+}
+
+export async function markStoryAsArchived(
+  mongo: Db,
+  tenantID: string,
+  storyID: string,
+  now: Date
+) {
+  const result = await collection(mongo).findOneAndUpdate(
+    { id: storyID, tenantID },
+    {
+      $set: {
+        isArchiving: false,
+        isArchived: true,
+        archivedAt: now,
```

Comment:
It would seem like that's a good idea at first glance. However, we use the unarchivedAt as a filter to prevent auto-archiving from running on stories that admin's specifically decide they want to unarchive. Otherwise, whenever they unarchive a story, the auto-archiver would just archive it again in an hour or two.